ActionsForCAP/gap/ActionsCategory.gi

```diff
@@ -522,6 +523,10 @@ InstallGlobalFunction( ADD_FUNCTIONS_FOR_LEFT_AND_RIGHT_ACTIONS_CATEGORY,
     AddIsEqualForCacheForMorphisms( category,
       IsIdenticalObj );
     
+    underlying_category := UnderlyingCategory( category );
+    
+    if CanCompute( underlying_category, "IsCongruentForMorphisms" ) then
```

Comment:
There is a serious change in the behaviour of the action category due to this line.

If `underlying_category` happens to be not computable (i.e., no function for `IsCongruentForMorphisms`),
then `IsEqualForObjects` will be set to `ReturnFail`, making the proper functioning of the action category a gamble unless you don't set its caches to strict. Maybe your error that you mentioned in #410 is due to this change (which I assume you have already implemented locally).

Two possible tests:
1) Test if your bug disappears once you set the caching to strict
2) Test if your bug disappears if you change the notion of equality of objects in `ActionCategory` in a way that it relies on `IsEqualForMorphisms` instead of `IsCongruentForMorphisms` in the underlying category. I'm not sure yet if this is a proper solution, since I'm not sure why I haven't chosen this notion from the very beginning.

Reply:
The error did not disappear with the first test: `SetCachingOfCategoryCrisp( underlying_category )`.

Reply:
The error did not disappear with the second test:
```
    if CanCompute( underlying_category, "IsEqualForMorphisms" ) then

    ##
    AddIsEqualForObjects( category,
      function( action_object_1, action_object_2 )

        return IsEqualForObjects( ActionDomain( action_object_1 ), ActionDomain( action_object_2 ) )
               and
               IsEqualForMorphisms( StructureMorphism( action_object_1 ), StructureMorphism( action_object_2 ) );

    end );
```

Reply:
The first test is wrong. You don't set the caching of the underlying category to crisp, but of the category that you are manipulating in this PR, which is the action category.

Reply:
The error did not disappear after adding:

`SetCachingOfCategoryCrisp( action_category )`

Reply:
Then I'll need to actually see that error eventually.

Nevertheless, `SetCachingOfCategoryCrisp( action_category )` will be mandatory to add in the case where
`    if CanCompute( underlying_category, "IsCongruentForMorphisms" ) then`
yields `false.`

Reply:
This last comment should be added to the action category constructors. What do you think?

Reply:
Yes, you can add it there.

Reply:
Any news concerning this PR?
The only missing point seems still to be the caching, which should be set to crisp in the non-computable case.

Reply:
First of all the original title of this PR is by now misleading and should be changed.

Still, one of the points here is that `AddIsEqualForObjects` calls
```
IsCongruentForMorphisms( StructureMorphism( action_object_1 ), StructureMorphism( action_object_2 ) );
```
which makes no sense if `CanCompute( underlying_category, "IsCongruentForMorphisms" ) = false`.

The second point is that in this case the explicit call of `AddIsCongruentForMorphisms` should also be prevented.

Reply:
I agree that `IsEqualForObjects` must be there. Could you please make a suggestion?

Reply:
If the underlying category is not computable, then do the following 2 steps for the action category:
- set `IsEqualForObject` and `IsEqualForMorphisms` to `IsIdenticalObj`  
- set the caching to crisp.